File: baboon-compiler/src/main/resources/baboon-runtime/python/baboon_exceptions.py


class BaboonCodecException(Exception):
    @staticmethod
    def ConversionNotFound(msg: str) -> 'BaboonCodecException':
        return BaboonCodecException(f"ConversionNotFound: {msg}")

    @staticmethod
    def CodecNotFound(msg: str) -> 'BaboonCodecException':
        return BaboonCodecException(f"CodecNotFound: {msg}")

    @staticmethod
    def EncoderFailure(msg: str, cause: Exception = None) -> 'BaboonCodecException':
        exc = BaboonCodecException(f"EncoderFailure: {msg}")
        exc.__cause__ = cause
        return exc

    @staticmethod
    def DecoderFailure(msg: str, cause: Exception = None) -> 'BaboonCodecException':
        exc = BaboonCodecException(f"DecoderFailure: {msg}")
        exc.__cause__ = cause
        return exc

    @staticmethod
    def ConverterFailure(msg: str, cause: Exception = None) -> 'BaboonCodecException':
        exc = BaboonCodecException(f"ConverterFailure: {msg}")
        exc.__cause__ = cause
        return exc

File: baboon-compiler/src/main/resources/baboon-runtime/cs/BaboonExceptions.cs
#nullable enable

using System;

// ReSharper disable UnusedTypeParameter
// ReSharper disable CheckNamespace
// ReSharper disable UnusedAutoPropertyAccessor.Global
// ReSharper disable MemberCanBeProtected.Global
// ReSharper disable MemberCanBePrivate.Global
// ReSharper disable ConvertToPrimaryConstructor
// ReSharper disable UnusedMember.Global
// ReSharper disable UnusedMemberInSuper.Global
// ReSharper disable ArrangeNamespaceBody
// ReSharper disable UnusedType.Global
// ReSharper disable InconsistentNaming
// ReSharper disable ClassCanBeSealed.Global

namespace Baboon.Runtime.Shared
{
    public abstract class BaboonCodecException : Exception
    {
        protected BaboonCodecException(string message) : base(message)
        {
        }

        protected BaboonCodecException(string message, Exception? cause) : base(message, cause)
        {
        }

        public sealed class EncoderFailure : BaboonCodecException
        {
            public EncoderFailure(string message) : base(message)
            {
            }

            public EncoderFailure(string message, Exception? cause) : base(message, cause)
            {
            }
        }

        public sealed class DecoderFailure : BaboonCodecException
        {
            public DecoderFailure(string message) : base(message)
            {
            }

            public DecoderFailure(string message, Exception? cause) : base(message, cause)
            {
            }
        }

        public sealed class ConverterFailure : BaboonCodecException
        {
            public ConverterFailure(string message) : base(message)
            {
            }

            public ConverterFailure(string message, Exception? cause) : base(message, cause)
            {
            }
        }

        public sealed class CodecNotFound : BaboonCodecException
        {
            public CodecNotFound(string message) : base(message)
            {
            }
        }

        public sealed class ConversionNotFound : BaboonCodecException
        {
            public ConversionNotFound(string message) : base(message)
            {
            }
        }
    }
}
